pkg/client/cache/reflector.go

```diff
@@ -333,11 +333,18 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 			if err != errorResyncRequested && err != errorStopRequested {
 				glog.Warningf("%s: watch of %v ended with: %v", r.name, r.expectedType, err)
 			}
-			return nil
+			if err != errorResyncRequested {
+				return nil
+			}
 		}
 		if r.canForceResyncNow() {
 			glog.V(4).Infof("%s: next resync planned for %#v, forcing now", r.name, r.nextResync)
-			return nil
+			if err := r.store.Resync(); err != nil {
+				return err
+			}
+			cleanup()
+			resyncCh, cleanup = r.resyncChan()
+			defer cleanup()
```

Comment:
Removed it.  Easy enough to add back if there's a problem, and like you said there will be other issues to resolve at that point.